perfkitbenchmarker/benchmark_spec.py

```diff
@@ -250,7 +250,14 @@ def Prepare(self):
 
   def Provision(self):
     """Prepares the VMs and networks necessary for the benchmark to run."""
-    vm_util.RunThreaded(lambda net: net.Create(), self.networks.values())
+    # Sort networks into a guaranteed order of creation based on dict key.
+    # There is a finite limit on the number of threads that are created to
+    # provision networks. Until support is added to provision resources in an
+    # order based on dependencies, this key ordering can be used to avoid
+    # deadlock by placing dependent networks later and their dependencies
+    # earlier.
```

Comment:
Thanks. Would you mind pointing out that the AWS network in particular requires this?